### PR TITLE
Add Jupyter notebook examples

### PR DESCRIPTION
**Pull request by antimora**
### Changes

Added Jupyter notebook examples with a readme file.

### Testing

Ran through the installation instructions and executed notebooks. 

### Screenshot

image

### CODE REVIEW

File: examples/notebook/plots.ipynb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   "source": [
+    "// Import packages\n",
+    "use burn::tensor::Tensor;\n",
+    "use burn_ndarray::NdArrayBackend;\n",
+    "\n",
+    "// Import plotting library\n",
+    "use evcxr_image::ImageDisplay;\n",
+    "\n",
+    "// Type alias for the backend\n",
+    "type B = NdArrayBackend<f32>;"
```
Comment (nathanielsimard) [inline examples/notebook/plots.ipynb:51]: I'm not sure Burn is actually used in this notebook, maybe we should remove the dependencies to make the compilation quicker?

**Review by nathanielsimard — COMMENTED** [review]
Very very cool 👏 It's nice to have this as an example showing what the Rust ecosystem has to offer to ML folks!

File: examples/notebook/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+To execute the Rust code within the notebooks, you must install the Evcxr kernel. Here's how to get
+started:
+
+### Install Evcxr Kernel
+
+1. **Build Evcxr Kernel:** Install the required package with the following command:
+
+   ```shell
+   cargo install --locked evcxr_jupyter
```
[the remark this answers is not in the document]
Comment (antimora) [reply examples/notebook/README.md:30]: @nathanielsimard I am not sure but here what the documentation says: The --locked flag can be used to force Cargo to use the packaged Cargo.lock file if it is available.

I blindly copied this from `Evcxr` site =) . Should I remove it?

File: examples/notebook/plots.ipynb (hunk 1 above)
Comment (antimora) [reply examples/notebook/plots.ipynb:51]: I want to come back and use it. I added a TODO. Lets leave it for now..

File: examples/notebook/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (antimora) [reply examples/notebook/README.md:30]: OK. I have removed it.

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**